### PR TITLE
Rebase on upstream dbd41db16a0e91b2566820898a3ab2d7dad4fe00

### LINKED ISSUE #85: rebase on bitcoin-core/secp256k1
Hi

I would have done a PR, but I was not sure how you handle rebasing. 

Anyway, it would be great to rebase this on upstream. I am particularly interested in pulling in this change: https://github.com/bitcoin-core/secp256k1/pull/337, so I can skip some custom commits on my fork of libwally-core.

### PR DESCRIPTION
**Pull request by jonasnick**
Fixes #85. My personal motivation is that I'd like to add a few modules soon that would benefit from the latest upstream changes, but are more appropriate to add here than upstream (such as [ECDSA adaptor sigs](https://github.com/jonasnick/secp256k1/pull/14) @nkohen).

This PR was created by running `git rebase upstream/master`.
The only conflicts I had to fix were

1. related to the upstream addition of the iters argument to the benchmark interface
2. moving config arguments from travis.yml to travis.sh

See [this branch](https://github.com/jonasnick/secp256k1-zkp/commits/secp256k1-rebase-backup2) for the fixups I applied).
Should be easy to review/recreate @apoelstra @real-or-random @benma

### CODE REVIEW

File: src/scalar_4x64_impl.h [hunk 1]
```diff
@@ -946,4 +956,102 @@ SECP256K1_INLINE static void secp256k1_scalar_mul_shift_var(secp256k1_scalar *r,
...
+
+        over1 = secp256k1_scalar_check_overflow(r1);
+        over2 = secp256k1_scalar_check_overflow(r2);
+        over_count++;
+   } while (over1 | over2);
+}
+
+#undef ROTL32
+#undef QUARTERROUND
+#undef LE32
```
Comment (benma) [inline src/scalar_4x64_impl.h:1055]: `#undef BE32` missing?

File: src/scalar_8x32_impl.h [hunk 2]
```diff
@@ -718,4 +731,114 @@ SECP256K1_INLINE static void secp256k1_scalar_mul_shift_var(secp256k1_scalar *r,
...
+
+        over1 = secp256k1_scalar_check_overflow(r1);
+        over2 = secp256k1_scalar_check_overflow(r2);
+        over_count++;
+   } while (over1 | over2);
+}
+
+#undef ROTL32
+#undef QUARTERROUND
+#undef LE32
```
Comment (benma) [inline src/scalar_8x32_impl.h:842]: `#undef BE32` missing?

File: src/scalar_4x64_impl.h (hunk 1 above)
Comment (jonasnick) [reply src/scalar_4x64_impl.h:1055]: Can't find a `BE32`  in this file

**Comment by jonasnick** [comment]
Thanks for the review!

> There are conflicts with the target branch though.

I think that's to be expected. This branch will become master and the previous master will be renamed to something else (https://github.com/ElementsProject/secp256k1-zkp/issues/68#issuecomment-498351737).

File: src/scalar_4x64_impl.h (hunk 1 above)
Comment (benma) [reply src/scalar_4x64_impl.h:1055]: Indeed, was removed in https://github.com/ElementsProject/secp256k1-zkp/commit/cb8f059724ea58cdf90e3a9637625aa2e60259c7, so it was a rebase error on my side.

**Comment by jonasnick** [comment]
Didn't push the branch in this PR to the secp256k1-zkp branch but instead pushed https://github.com/real-or-random/secp256k1-zkp/tree/202007-secp256k1-rebase. @real-or-random did the same recreated the same history but cleaned up the history a bit and rebased on a newer version of upstream libsecp. Closing.

**Comment by jonasnick** [comment]
The pre force push `secp256k1-zkp` branch is archived at https://github.com/ElementsProject/secp256k1-zkp/tree/secp256k1-zkp-2020-07-24.

**Comment by real-or-random** [comment]
> Didn't push the branch in this PR to the secp256k1-zkp branch but instead pushed https://github.com/real-or-random/secp256k1-zkp/tree/202007-secp256k1-rebase. @real-or-random did the same recreated the same history but cleaned up the history a bit and rebased on a newer version of upstream libsecp. Closing.

post-(merge ?) ACK

**Pull request closed without merging**